templates/install_puppet.bat.erb

```diff
@@ -0,0 +1,12 @@
+:wait_for_pid
+timeout /t 5 /nobreak > NUL
+FOR /F "tokens=*" %%A IN ('tasklist /FI "PID eq <%= @puppet_agent_pid %>"  /NH') DO set _task=%%A
+echo %_task% | findstr "No tasks are running" >nul
+IF NOT %errorlevel% == 0 ( GOTO wait_for_pid )
+
+
+msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*v "<%= @_logfile %>"
```

Comment:
I have not had @cowofevil test it yet. I have been running it against my servers from a master and rolling back.